doc/exts/pylint_options.py

```diff
@@ -90,7 +107,13 @@ def _create_checker_section(
             continue
 
         # Get current value of option
-        value = getattr(linter.config, option.name.replace("-", "_"))
+        if (
+            option.name not in DYNAMICALLY_DEFINED_OPTIONS
+            or "default" not in DYNAMICALLY_DEFINED_OPTIONS[option.name]
+        ):
+            value = getattr(linter.config, option.name.replace("-", "_"))
+        else:
+            value = DYNAMICALLY_DEFINED_OPTIONS[option.name]["default"]
```

Comment:
```suggestion
        try:
            value = DYNAMICALLY_DEFINED_OPTIONS[option.name]["default"]
        except KeyError:
            value = getattr(linter.config, option.name.replace("-", "_"))
```

Would that make sense and be a bit easier to read?

Reply:
Make sense ! I'll rebase and ask you to approve again :D